feat(scheduler-targets-alpha): `KinesisStreamPutRecord` Target

go-to-k opened the pull request:
This PR adds KinesisStreamPutRecord Target for EventBridge Scheduler.

Closes #27451.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation reviewed (dismissed):
The pull request linter has failed. See the aws-cdk-automation comment below for failure reasons. If you believe this pull request should receive an exemption, please comment and provide a justification.


A comment requesting an exemption should contain the text `Exemption Request`. Additionally, if clarification is needed add `Clarification Request` to a comment.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to a README file.
	❌ Features must contain a change to a test file.
	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to a README file.
	❌ Features must contain a change to a test file.
	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to a README file.
	❌ Features must contain a change to a test file.
	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 6f7ad358fef559216e0a8d3a71aa5581f53e3054
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6066376c-c8b8-4f4c-b878-f986b6d598ca%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: f24b427ef89773b993b7e73263af68506fcca461
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bfbf7c78-129b-446d-94a1-f6bcee41e096%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to a README file.
	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 2d52200609c8e8092cf2e6ebe5962fcdd0237c06
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=26d91844-e54f-4586-9927-dda0d88e20f7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 40d4e14ce255f5a16243999d10f376f5ea694c76
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0327f238-ec31-40b9-9895-d5fc4f8f0ba4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 441bc6fc150a72fad3058dbccd6fe1f1f136efe6
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4b293d8f-0927-4d44-83a5-622dc75daeff%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: c8fe724c21ef33d57bbdb16e38390fed957a557b
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3dd990ee-c00a-4716-8be2-f8912875f03a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: e6f793418134f5f2ac758bba71964098b178845b
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=96c374af-84f2-4365-8715-9888595bd94c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: dbf89189867bb49dd949de04b2be9901ea564ec8
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c8242cba-e5d5-4a16-bf5b-3d6af9f7090e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 5c6ada0d12b7cc96822bdaf1c5b604ae1d922227
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=582e2df2-f84e-47eb-8854-59145373125a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 7d0450af027b0bc303320f14ba7c9b3ce26d790a
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=33265e80-9e05-402d-96df-87e378078cbe%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: b963d7edc956ef7aacf79b47e48244a1560433d5
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a066bc4b-be55-4d7c-8b20-7a7c0dc2f5d8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

lpizzinidev commented on packages/@aws-cdk/aws-scheduler-targets-alpha/lib/kinesis-stream-put-record.ts:
```suggestion
   * The length must be between 1 and 256.
```

lpizzinidev commented on packages/@aws-cdk/aws-scheduler-targets-alpha/test/kinesis-stream-put-record.test.ts:
```suggestion
  test.each([
    ['account', 'arn:aws: kinesis:us-east-1:999999999999:stream/Foo', /Both the schedule and the stream must be in the same account./],
    ['region', 'arn:aws: kinesis:eu-central-1:123456789012:stream/Foo', /Both the schedule and the stream must be in the same region./],
  ])('throws when Kinesis Data Stream is imported from different %s', (_, arn: string, expectedError: RegExp) => {
    const importedStream = kinesis.Stream.fromStreamArn(stack, 'ImportedStream', arn);
    const streamTarget = new KinesisStreamPutRecord(importedStream, {
      partitionKey: 'key',
    });
    expect(() =>
      new Schedule(stack, 'MyScheduleDummy', {
        schedule: expr,
        target: target,
      })).toThrow(expectedError);
  });
```
`fromStreamArn` should use the same stack.
[`fromStreamAttributes`](https://github.com/aws/aws-cdk/blob/02957dfddd895ffcca1dd485da312544babf6ccb/packages/aws-cdk-lib/aws-kinesis/lib/stream.ts#L738) will require to solve the environment for the imported stream ARN:
```
  public static fromStreamAttributes(scope: Construct, id: string, attrs: StreamAttributes): IStream {
    class Import extends StreamBase {
      public readonly streamArn = attrs.streamArn;
      public readonly streamName = Stack.of(scope).splitArn(attrs.streamArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName!;
      public readonly encryptionKey = attrs.encryptionKey;
    }

    return new Import(scope, id, {
      environmentFromArn: attrs.streamArn,
    });
  }
```
Also, can you please add unit tests to verify that the environment gets resolved correctly when using `fromStreamArn` and `fromStreamAttributes` after this change?

lpizzinidev reviewed (changes requested):
Looks good overall, thanks 👍
The unit tests on imported streams will require a fix on the importing function.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 945604f6ff48c5a5ec8ce61874bd9a8a360b6061
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f263d778-50b8-4f0b-946d-bbeeb4bba043%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

go-to-k commented:
@lpizzinidev 

I just changed! Please check them.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 81472588bf04fb4ca6748f00ed327436d095b7aa
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d7f8d00f-8967-407b-b1ac-a81bb9ba6a14%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 1c99fd601302549a1b55c2311a7114aa666f8c64
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=676c64a4-1e43-404b-a74c-f8f1ecd101eb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

lpizzinidev reviewed (approved):
Thanks 👍

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 77e409ec01eaafbe3d7d03f429276df03df82227
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=42328fbf-7a8a-4f79-b579-f25018b80317%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
This PR cannot be merged because it has conflicts. Please resolve them. The PR will be considered stale and closed if it remains in an unmergeable state.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: ed8b2ddcace3c47eb744d682036f6c153c685b73
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c7ff45d5-3217-4a19-b5a2-210ed7da7861%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

vinayak-kukreja reviewed (dismissed):
Hey @go-to-k , thank you for this contribution.

go-to-k commented:
@vinayak-kukreja 

Thanks for your approval! I fixed conflicts. Please approve again.

I fixed [the other PR](https://github.com/aws/aws-cdk/pull/27842) the same way, but when one of them gets merged, the other one gets conflicts again. If that happens, I will fix it again.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 8498c8aa127ea4d5a65b48117300420ebdf0a104
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e2310200-f721-4100-9a9a-7ce24864c537%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d47f95875e7dcc3d666e5ba192d03ebf45e5d838
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=361ec4b7-8eac-4619-8eb3-7a121a67ac8e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
